src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile_strategy.py

```diff
@@ -233,3 +236,19 @@ def extra_compile_classpath_iter():
           yield (conf, jar)
 
     return list(extra_compile_classpath_iter())
+
+  # Work in a tmpdir so we don't stomp the main analysis files on error.
+  # The tmpdir is cleaned up in a shutdown hook, because background work
+  # may need to access files we create there even after this method returns.
+  def ensure_analysis_tmpdir(self):
+    # Do this lazily, so we don't trigger creation of a worker pool unless we need it.
+    if not os.path.exists(self.analysis_tmpdir()):
+      os.makedirs(self.analysis_tmpdir())
+      if self._delete_scratch:
+        self.context.background_worker_pool().add_shutdown_hook(
+          lambda: safe_rmtree(self.analysis_tmpdir()))
+
+  @abstractproperty
+  def analysis_tmpdir(self):
+    """Post processes VTS that have been fetched from the cache."""
+    pass
```

Comment:
no need for 'pass' in an abstract property if you have a pydoc string. The body never gets executed.